pkg/kubelet/kuberuntime/kuberuntime_gc.go

```diff
@@ -356,9 +356,33 @@ func (cgc *containerGC) evictPodLogsDirectories(allSourcesReady bool) error {
 	logSymlinks, _ := osInterface.Glob(filepath.Join(legacyContainerLogsDir, fmt.Sprintf("*.%s", legacyLogSuffix)))
 	for _, logSymlink := range logSymlinks {
 		if _, err := osInterface.Stat(logSymlink); os.IsNotExist(err) {
+			if containerID, err := getContainerIDFromLegacyLogSymlink(logSymlink); err == nil {
+				status, err := cgc.manager.runtimeService.ContainerStatus(containerID)
+				if err != nil {
+					klog.Infof("Error getting ContainerStatus for containerID %q: %v", containerID, err)
```

Comment:
This means if we can't get the container status (e.g. container runtime is down), then we remove the symlink.  Does this re-introduce the bug you were trying to fix?

Reply:
runtimeService.ContainerStatus() returning error, in most cases, means that the container is gone.
We should treat such error the same as State == runtimeapi.ContainerState_CONTAINER_EXITED.

Reply:
Ah, right.  Can you add a TODO saying that we should handle container not found (i.e. container was deleted) differently from other errors once we have https://github.com/kubernetes/kubernetes/issues/63336

Reply:
@dashpole 
I have added the TODO